Holyhead-Harpies/Bangazon-Ruby-on-Rails-Web-Interface
# Jw user auth

## Status
**READY**

## Migrations
YES

## Description
User authorization functionality:
1. User can sign up -> adds new user to Users table
2. User can log in -> creates session, nav bar functions are shown
3. USer can log out -> destroys session, returns to home page

## Related PRs
None

## Todos


## Deploy Notes
Fetch and checkout branch
```
git fetch origin jw-user-auth
git checkout jw-user-auth
```
Migrate
```
rails db:rollback
rails db:migrate
```
Run unit tests
```
rails test
```
All tests should pass.

Run the server
```
rails server
```

Manual testing - 
1. Go to `http://localhost:3000`, navbar should show only the logo.
2. Click 'Sign Up', browser should take you to sign up form.
3. Submit form without fields filled, error message should flash on the page.
4. Submit form with a password less than 5 characters, error message should flash on the page.
5. Submit form with information and valid password, browser should redirect to home page, navbar menu items should appear, check database Users table, new user should appear with hash in password_digest column.
6. Click 'Log Out' from navbar, menu items should disappear and the browser should redirect to home page. 
7. Click 'Log In' and submit without credentials, error message should appear and browser will not redirect. 
8. Click the logo, browser should redirect and error message should disappear.
9. Click 'Log In' and submit with correct credentials, navbar menu items should appear, browser should redirect to home page.
10. Click 'Log Out' from navbar, menu items should disappear and the browser should redirect to home page. 

## Steps to Test or Reproduce
```
rails test
```
All tests should pass


## Impacted Areas in Application
List general components of the application that this PR will affect:

* This PR will make session user id available to other parts of the application
